algorithms/kernel/svm/oneapi/svm_train_thunder_oneapi_impl.i

```diff
@@ -0,0 +1,277 @@
+
+    KernelRange localRange(nWS);
+    KernelRange globalRange(nWS);
+
+    KernelNDRange range(1);
+    range.global(globalRange, &status);
+    DAAL_CHECK_STATUS_VAR(status);
+    range.local(localRange, &status);
+    DAAL_CHECK_STATUS_VAR(status);
+
+    context.run(range, kernel, args, &status);
+    DAAL_CHECK_STATUS_VAR(status);
+
+    return status;
+}
+
+template <typename algorithmFPType, typename ParameterType>
+bool SVMTrainOneAPI<algorithmFPType, ParameterType, thunder>::checkStopCondition(const algorithmFPType diff, const algorithmFPType diffPrev,
+                                                                                 const algorithmFPType eps, const size_t nNoChanges,
+                                                                                 size_t & sameLocalDiff)
+{
+    sameLocalDiff = utils::internal::abs(diff - diffPrev) < eps * 1e-2 ? sameLocalDiff + 1 : 0;
+
+    if (sameLocalDiff > nNoChanges || diff < eps)
+    {
+        return true;
+    }
+    return false;
+}
+
+template <typename algorithmFPType, typename ParameterType>
+services::Status SVMTrainOneAPI<algorithmFPType, ParameterType, thunder>::compute(const NumericTablePtr & xTable, NumericTable & yTable,
+                                                                                  daal::algorithms::Model * r, const ParameterType * svmPar)
+{
+    services::Status status;
+
+    auto & context    = services::Environment::getInstance()->getDefaultExecutionContext();
+    const auto idType = TypeIds::id<algorithmFPType>();
+
+    const algorithmFPType C(svmPar->C);
+    const algorithmFPType eps(svmPar->accuracyThreshold);
+    const algorithmFPType tau(svmPar->tau);
+    const size_t maxIterations(svmPar->maxIterations);
+    const size_t cacheSize(svmPar->cacheSize);
+    kernel_function::KernelIfacePtr kernel = svmPar->kernel->clone();
+
+    const size_t nVectors  = xTable->getNumberOfRows();
+    const size_t nFeatures = xTable->getNumberOfColumns();
+    // ai = 0
+    auto alphaU = context.allocate(idType, nVectors, &status);
+    context.fill(alphaU, 0.0, &status);
+    DAAL_CHECK_STATUS_VAR(status);
+    auto alphaBuff = alphaU.template get<algorithmFPType>();
+
+    auto maskBuff = context.allocate(idType, nVectors, &status);
+
+    BlockDescriptor<algorithmFPType> yBD;
+    DAAL_CHECK_STATUS(status, yTable.getBlockOfRows(0, nVectors, ReadWriteMode::readOnly, yBD));
+    auto yBuff = yBD.getBuffer();
+
+    // gradi = -yi
+    auto gradU = context.allocate(idType, nVectors, &status);
+    DAAL_CHECK_STATUS_VAR(status);
+    auto gradBuff = gradU.template get<algorithmFPType>();
+
+    DAAL_CHECK_STATUS(status, Helper::makeInversion(yBuff, gradBuff, nVectors));
+
+    TaskWorkingSet<algorithmFPType> workSet(nVectors);
+
+    DAAL_CHECK_STATUS(status, workSet.init());
+
+    const size_t nWS        = workSet.getSize();
+    const size_t nNoChanges = 5;
+    const size_t innerMaxIterations(nWS * 1000);
```

Comment:
A lot of magic number over there, I'd recomment to put their declaration to `SVMTrainOneAPI` as `static` with relevant names and short comment. It's very difficult to understand the meaning of `nNoChanges` and why `innerMaxIteration` is 1000 times `nWS` if the reader is not familiar with the algorithms in details.

Reply:
Done